File: launcher/minimal/src/main/java/ink/reactor/launcher/logger/NoneLogger.java
package ink.reactor.launcher.logger;

import ink.reactor.kernel.logger.Logger;
import ink.reactor.kernel.logger.LoggerFormatter;

public final class NoneLogger implements Logger {
    public LoggerFormatter getLoggerFormatter() {
        return (s, _) -> s;
    }
    public void debug(final String message) {}
    public void log(final String message) {}
    public void info(final String message) {}
    public void warn(final String message) {}
    public void error(final String message) {}
    public void error(final String message, final Throwable throwable) {}
}

File: launcher/minimal/src/main/kotlin/ink/reactor/launcher/logger/type/NoneLogger.kt
package ink.reactor.launcher.logger.type

import ink.reactor.kernel.logger.Logger
import ink.reactor.microkernel.logger.JavaLoggerFormatter

class NoneLogger(
    override val loggerFormatter: JavaLoggerFormatter = JavaLoggerFormatter()
) : Logger {
    override fun debug(message: String) {}
    override fun log(message: String) {}
    override fun info(message: String) {}
    override fun warn(message: String) {}
    override fun error(message: String) {}
    override fun error(message: String, throwable: Throwable) {}
}
